orion/handlers/http/http.go

```diff
@@ -204,20 +204,15 @@ func (h *httpHandler) serialize(ctx context.Context, msg proto.Message) ([]byte,
 			}
 		}
 	}
-	switch serType {
-	case modifiers.JSONPB:
-		sData, err := h.mar.MarshalToString(msg)
-		return []byte(sData), ContentTypeJSON, err
-	case modifiers.ProtoBuf:
+	if serType == modifiers.ProtoBuf {
 		data, err := proto.Marshal(msg)
 		return data, ContentTypeProto, err
-	case modifiers.JSON:
-		fallthrough
-	default:
-		// modifiers.JSON goes in here
-		data, err := json.Marshal(msg)
-		return data, ContentTypeJSON, err
+	} else if serType == modifiers.JSONPB || h.config.DefaultJSONPB {
```

Comment:
I think there's a disparity in behavior here. If the default is `jsonpb`, and in my handler I set `modifiers.SerializeOutJSON(ctx)`, then I expect my response to be serialized using the `json` package, not `jsonpb`.

https://github.com/carousell/Orion/blob/f006d565d9ed9d1d28b35f702affc1faade479d0/orion/handlers/http/http.go#L193-L206

You can replace L203 with

```
if h.config.DefaultJSONPB {
  serType = modifiers.JSONPB
} else {
  serType = modifiers.JSON
}
```

then I don't think you need to change this switch-case block.

Reply:
yes good point - actually we also want to set `serType = modifiers.JSONPB` in the case that the `Accept` or `Content-Type` header is set to `"application/json"`. I've updated the code with something similar to your suggestion